sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -294,23 +294,25 @@ class Dataset[T] private[sql](
     // We set a minimum column width at '3'
     val minimumColWidth = 3
 
+    //Regular expression matching full width characters
+    val fullWidthRegex = """[\u1100-\u115F\u2E80-\uA4CF\uAC00-\uD7A3\uF900-\uFAFF\uFE10-\uFE19\uFE30-\uFE6F\uFF00-\uFF60\uFFE0-\uFFE6]""".r
```

Comment:
I wonder if detecting this using a regex for each character is slow? probably not meaningfully enough to care about, but I wonder.

Reply:
I generated 1000 strings, each consisting of 1000 characters with a random unicode of 0x0000-0xFFFF. (a total of 1 million characters.)
Then use this regular expression to find the full width character of these strings. 
I tested 100 rounds and then averaged.
It takes 49 milliseconds to complete matching all 1000 strings.